sycl/source/handler.cpp

```diff
@@ -417,6 +417,42 @@ void handler::extractArgsAndReqs() {
 
   const bool IsKernelCreatedFromSource = MKernel->isCreatedFromSource();
 
+  // TODO: to do that in compile time.
+  size_t real_KernelArgsNum = UnPreparedArgs.size();
+  for (size_t I = 0; I < UnPreparedArgs.size(); ++I) {
+    const detail::kernel_param_kind_t &Kind = UnPreparedArgs[I].MType;
+    if (Kind == detail::kernel_param_kind_t::kind_accessor) {
+      const int &Size = UnPreparedArgs[I].MSize;
+      // For args kind of accessor Size is information about accessor.
+      // The first 11 bits of Size encodes the accessor target.
+      const access::target AccTarget =
+          static_cast<access::target>(Size & 0x7ff);
+      switch (AccTarget) {
+      case access::target::global_buffer:
+      case access::target::constant_buffer:
+      case access::target::local: {
+        if (!IsKernelCreatedFromSource) {
+          real_KernelArgsNum += 3;
+        }
+        break;
+      }
+      case access::target::image:
+      case access::target::image_array:
+      case access::target::host_image:
+      case access::target::host_buffer: {
+        break;
+      }
+      }
+    } else if (Kind == detail::kernel_param_kind_t::kind_stream) {
+      if (!IsKernelCreatedFromSource) {
+        real_KernelArgsNum += 12;
```

Comment:
Do we really need to "reserve" pricise amount and not worst case estimate like 12*num-of-args?
We can reclaim the extra memory with a subsequent "shrink_to_fit".

Reply:
Of course not, I like your comment, I have to check it out.

Reply:
What if you just don't do "shrink_to_fit" and waste some small memory for some time? How long living is this vector?

Reply:
It is possible, 
I checked this case, it doesn't affect performance. Lifetime matches with the handler class, that is, this vector lives for a short time, and then it will be deleted.
Should we go this way then?

Reply:
I'd prefer that, yes.